source/Applications/Advanced/TransformPointCloudViaArucoMarker/TransformPointCloudViaArucoMarker.cpp

```diff
@@ -0,0 +1,313 @@
+        }
+
+        cv::Mat newRotation = rotation.inv();
+        cv::Mat newTranslation = -newRotation * translation;
+
+        auto cvInverted = cv::Mat(4, 4, CV_32F, 0.0);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                cvInverted.at<float>(i, j) = newRotation.at<float>(i, j);
+            }
+        }
+        for(int i = 0; i < 3; ++i)
+        {
+            cvInverted.at<float>(i, 3) = newTranslation.at<float>(i, 0);
+        }
+        cvInverted.at<float>(3, 3) = 1.0;
+
+        auto *ptr = &cvInverted.at<float>(0, 0);
+        auto invertedTransform = Zivid::Matrix4x4(ptr, ptr + 16);
+
+        return invertedTransform;
+    }
+
+	cv::Mat pointCloudToColorBGR(const Zivid::PointCloud &pointCloud)
+    {
+        auto rgb = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        pointCloud.copyData(
+            reinterpret_cast<Zivid::ColorRGBA *>(rgb.data)); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+        auto bgr = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        cv::cvtColor(rgb, bgr, cv::COLOR_RGBA2BGR);
+
+        return bgr;
+    }
+
+	cv::Mat colorBGRToGray(const cv::Mat &bgr)
+	{
+        cv::Mat gray;
+        cv::cvtColor(bgr, gray, cv::COLOR_BGR2GRAY);
+        return gray;
+	}    
+
+	void displayBGR(const cv::Mat &bgr, const std::string &bgrName)
+    {
+        cv::namedWindow(bgrName, cv::WINDOW_AUTOSIZE);
+        cv::imshow(bgrName, bgr);
+        cv::waitKey(0);
+    }
+
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        const auto arucoMarkerFile = std::string(ZIVID_SAMPLE_DATA_DIR) + "/ArucoMarkerInCameraOrigin.zdf";
+        std::cout << "Reading ZDF frame from file: " << arucoMarkerFile << std::endl;
+        const auto frame = Zivid::Frame(arucoMarkerFile);
+        auto pointCloud = frame.pointCloud();
+		
+		std::cout << "Converting to OpenCV image format" << std::endl;
+        const auto bgrImage = pointCloudToColorBGR(pointCloud);
+        const auto grayImage = colorBGRToGray(bgrImage);
+        
+        std::cout << "Configuring ArUco marker" << std::endl;
+        const auto markerDictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_4X4_100);
+        std::vector<int> markerIds;
+        std::vector<std::vector<cv::Point2f>> markerCorners;
+        cv::Ptr<cv::aruco::DetectorParameters> detectorParameters = cv::aruco::DetectorParameters::create();
+        detectorParameters->cornerRefinementMethod = cv::aruco::CORNER_REFINE_SUBPIX;
+
+        std::cout << "Detecting ArUco Marker" << std::endl;
+        cv::aruco::detectMarkers(grayImage, markerDictionary, markerCorners, markerIds, detectorParameters);
+
+		if(markerIds.empty())
+        {
+            std::cout << "No ArUco markers detected";
+            exit(EXIT_SUCCESS);
```

Comment:
Do you want to call this a "success"? Not sure about your conventions in the samples, but think about it at least.

Reply:
I need your input here. I don't know what is the best way to break from main here.
The first and the most basic option I tried was `return {}` but then I also thought `EXIT_SUCCESS` and `EXIT_FAILURE` might be better options. If there is no marker in the scene the code sample exits as it should, it is not unexpected behavior, which is why I thought `EXIT_SUCCESS` was a better option. Please correct me if I am wrong.

Reply:
I agree that you should do either `return EXIT_SUCCESS` or `return EXIT_FAILURE`. (no need to use `exit()`)

Which one of the two is up to you I think. Is "no marker" a failure? Maybe, maybe not. I'm fine with either. But please change to use return-statement instead of exit().

Reply:
Talked IRL, change to return EXIT_SUCCESS